src/js/components/List/__tests__/List-test.js

```diff
@@ -1,13 +1,157 @@
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.mouseOut(getByText('beta'));
+    expect(container.firstChild).toMatchSnapshot();
+    expect(onClickItem).toBeCalledTimes(0);
+  });
+
+  test('focus and blur', () => {
+    const onClickItem = jest.fn();
+    const { container, getByText } = render(
+      <Grommet>
+        <List
+          data={[{ a: 'alpha' }, { a: 'beta' }]}
+          onClickItem={onClickItem}
+        />
+      </Grommet>,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.focus(getByText('beta'));
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.blur(getByText('beta'));
+    expect(container.firstChild).toMatchSnapshot();
+    expect(onClickItem).toBeCalledTimes(0);
+  });
+
+  test('ArrowDown key', () => {
+    const onClickItem = jest.fn();
+    const { container, getByText } = render(
+      <Grommet>
+        <List
+          data={[{ a: 'alpha' }, { a: 'beta' }]}
+          onClickItem={onClickItem}
+        />
+      </Grommet>,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.click(getByText('alpha'));
+    fireEvent.mouseOver(getByText('alpha'));
+    fireEvent.keyDown(getByText('alpha'), {
+      key: 'ArrowDown',
+      keyCode: 40,
+      which: 40,
+    });
+    expect(onClickItem).toHaveBeenCalled();
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('ArrowDown key on last element', () => {
+    const onClickItem = jest.fn();
+    const { container, getByText } = render(
+      <Grommet>
+        <List
+          data={[{ a: 'alpha' }, { a: 'beta' }]}
+          onClickItem={onClickItem}
+        />
+      </Grommet>,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.click(getByText('beta'));
+    fireEvent.mouseOver(getByText('beta'));
+    fireEvent.keyDown(getByText('beta'), {
+      key: 'ArrowDown',
+      keyCode: 40,
+      which: 40,
+    });
+    expect(onClickItem).toHaveBeenCalled();
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('ArrowUp key', () => {
+    const onClickItem = jest.fn();
+    const { container, getByText } = render(
+      <Grommet>
+        <List
+          data={[{ a: 'alpha' }, { a: 'beta' }]}
+          onClickItem={onClickItem}
+        />
+      </Grommet>,
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.click(getByText('beta'));
+    fireEvent.mouseOver(getByText('beta'));
+    fireEvent.keyDown(getByText('beta'), {
+      key: 'ArrowUp',
+      keyCode: 38,
+      which: 38,
+    });
+    expect(onClickItem).toHaveBeenCalled();
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('Enter key', () => {
+    const onClickItem = jest.fn();
+    const { container, getByText } = render(
+      <Grommet>
+        <List
+          data={[{ a: 'alpha' }, { a: 'beta' }]}
+          onClickItem={onClickItem}
+        />
+      </Grommet>,
```

Comment:
Since this List component is repeated on all the new tests I wonder if we can have one instance of it and use it across all tests generically. thoughts?

Reply:
Yes, I changed the setup so that in the beforeEach() for the relevant tests the onClickItem and App (list component) are created. They still have to be recreated before each test though otherwise the props and number of times onClickItem is called would be affected by previous tests.